### PR TITLE
Performance improvements

### PR DESCRIPTION
**Pull request by grantspeelman**
Before

```bash
ruby 3.3.6 (2024-11-05 revision 75015d4c1f) [x86_64-linux]
Warming up --------------------------------------
              simple    36.000 i/100ms
          odata-duty    14.000 i/100ms
Calculating -------------------------------------
              simple    332.284 (± 9.3%) i/s    (3.01 ms/i) -      1.656k in   5.032172s
          odata-duty    138.532 (±10.8%) i/s    (7.22 ms/i) -    700.000 in   5.124592s

Comparison:
              simple:      332.3 i/s
          odata-duty:      138.5 i/s - 2.40x  slower
```

After

```bash
ruby 3.3.6 (2024-11-05 revision 75015d4c1f) [x86_64-linux]
Warming up --------------------------------------
              simple    39.000 i/100ms
          odata-duty    40.000 i/100ms
Calculating -------------------------------------
              simple    411.432 (±12.2%) i/s    (2.43 ms/i) -      2.028k in   5.025922s
          odata-duty    424.624 (± 6.6%) i/s    (2.36 ms/i) -      2.120k in   5.016987s

Comparison:
          odata-duty:      424.6 i/s
              simple:      411.4 i/s - same-ish: difference falls within error
```

### CODE REVIEW

File: lib/odata_duty/edms.rb [hunk 1]
```diff
@@ -50,6 +50,12 @@ def self.to_oas2(is_collection: false)
       OAS_TYPE
     end
 
+    def self.to_value(object, _context)
+      object&.to_str
+    rescue StandardError => e
+      raise InvalidValue, e.message
+    end
```
Comment (grantspeelman) [inline lib/odata_duty/edms.rb:57]: much more performant than what the base class does:

```ruby
    def self.to_value(*args)
      new(*args).__to_value
    end
```

Will cleanup all these classes in a future PR

File: lib/odata_duty/executor.rb [hunk 2]
```diff
@@ -20,14 +20,22 @@ def initialize(schema:, url:, context:, query_options:)
...
         @query_options = (query_options || {}).to_h
       end
+
+      def url_for(*args, **kwargs)
+        url_builder.url_for(*args, **kwargs)
+      end
+
+      def current
+        @current ||= {}
+      end
```
Comment (grantspeelman) [inline lib/odata_duty/executor.rb:39]: allows caching things for the current request

File: lib/odata_duty/executor.rb [hunk 3]
```diff
@@ -20,14 +20,22 @@ def initialize(schema:, url:, context:, query_options:)
...
-        super(context)
+      def initialize(url_builder, endpoint:, query_options: nil)
+        @url_builder = url_builder
         @endpoint = endpoint
         @query_options = (query_options || {}).to_h
       end
+
+      def url_for(*args, **kwargs)
+        url_builder.url_for(*args, **kwargs)
+      end
```
Comment (grantspeelman) [inline lib/odata_duty/executor.rb:35]: remove the `missing_method` check

**Review by grdmnt — APPROVED** [review]
Looks good

**Pull request merged**